### PR TITLE
Enforce paid schedule payment coverage

### LINKED ISSUE #1119: booking_payment_schedules.status can flip to 'paid' without a corresponding payments row, guaranteeing financial drift
## What I'm seeing

In a live admin DB, `booking_payment_schedules` rows for a booking sit in `status='paid'` while:

- `payments` joined via `invoices.booking_id` returns zero rows
- `payment_sessions` filtered on `booking_payment_schedule_id` returns zero rows
- the issued invoice for the same booking has `paid_cents = 0`, `balance_due_cents = total_cents`, `status = 'sent'`

So the schedule row is the only thing in the financial picture that thinks money arrived. Nothing in the audit ledger (`action_ledger_entries`) records *how* it got there, either — there's no `finance.booking_payment.create` or equivalent against that booking.

This may have come from a legacy backfill (the bookings predate the schedule rows by months), but the structural problem is that the data model permits it: there is no constraint and no service-layer invariant that says "a `booking_payment_schedules` row can only reach `paid` when a `payments` row covers it."

## Why it matters

Three voyant-native UIs on the same screen disagree because they each query a different source of truth:

- `BookingPaymentScheduleList` renders `schedule.status === 'paid'` → "Platit"
- The Plati / payments card lists `payments` → empty
- `BookingInvoicesCard` renders `invoice.paid_cents` / `balance_due_cents` → "Achitat —, Sold 330 EUR"

The operator sees three different answers and has no way to tell which is right.

## Suggested direction

A few options to discuss before anyone commits to one:

1. **Service-layer invariant**: in `@voyantjs/finance`, only allow `bookingPaymentScheduleService.markPaid` (or whichever entrypoint flips status) when invoked alongside a `payments` insert in the same transaction. Direct `update … set status = 'paid'` from outside the service should be discouraged in docs and ideally prevented (e.g. by routing all schedule mutations through a single function that takes a payment payload when transitioning to `paid`).
2. **DB-level check**: a `CHECK` constraint or trigger that disallows `status = 'paid'` unless a matching `payments` row exists.
3. **Reconciliation job**: a periodic / on-demand reconciler that flags drift (schedule says paid, payments don't add up). Useful as defense-in-depth even with (1) or (2).
4. **Backfill helper**: a documented script that, given a `booking_payment_schedule_id` claimed-paid with no payment, generates the missing `payments` row (or reverts the schedule to `pending`) and writes the corresponding ledger entry. Right now downstream apps have to hand-roll this themselves.

(2) is the strongest guarantee, (1) is the cheapest to ship. Happy to PR whichever direction lands well — let me know.

Related: #1116 and #1117 cover other invoice/schedule-list issues we hit on the same screen.

### PR DESCRIPTION
**Pull request by mihaipxm**
## Summary
- Require booking payment schedules to have linked completed payment coverage before they can be marked `paid`
- Materialize a `payments` row against an outstanding booking invoice when a schedule payment session completes
- Return validation errors for direct paid schedule mutations that would create drift

Closes #1119

## Verification
- `pnpm --filter @voyantjs/finance lint`
- `pnpm --filter @voyantjs/finance typecheck`
- `pnpm --filter @voyantjs/finance test`
- Push hook full test lane: 127 successful / 127 total

### CODE REVIEW

File: packages/finance/src/service.ts [hunk 1]
```diff
@@ -670,6 +670,85 @@ function assertPaymentCanSettleInvoice(invoiceCurrency: string, data: CreatePaym
...
+    "id" | "bookingId" | "amountCents" | "currency"
+  >,
+) {
+  const [coverage] = await db
+    .select({
+      total: sql<number>`coalesce(sum(${payments.amountCents}), 0)::int`,
+    })
+    .from(paymentSessions)
+    .innerJoin(payments, eq(paymentSessions.paymentId, payments.id))
+    .innerJoin(invoices, eq(payments.invoiceId, invoices.id))
```
[the remark this answers is not in the document]
Comment (mihaipxm) [reply packages/finance/src/service.ts]: Fixed in `f4875ee`: coverage now groups by distinct `payments.id` before summing, so multiple paid sessions pointing at the same payment cannot inflate coverage. I also added a regression test for that duplicate-session case.

File: packages/finance/tests/integration/routes.test.ts [hunk 2]
```diff
@@ -649,12 +653,20 @@ describe.skipIf(!DB_AVAILABLE)("Finance routes", () => {
         .from(bookingPaymentSchedules)
         .where(eq(bookingPaymentSchedules.id, schedule.id))
       expect(storedSchedule?.status).toBe("paid")
+      const [payment] = await db.select().from(payments).where(eq(payments.id, data.paymentId))
```
[the remark this answers is not in the document]
Comment (mihaipxm) [reply packages/finance/tests/integration/routes.test.ts:657]: Fixed in `f4875ee`: the completion response is parsed before using `data.paymentId`.
